src/H5VLcallback.c

```diff
@@ -4703,30 +4704,20 @@ H5VL__link_create(H5VL_link_create_type_t create_type, void *obj, const H5VL_loc
  *-------------------------------------------------------------------------
  */
 herr_t
-H5VL_link_create(H5VL_link_create_type_t create_type, const H5VL_object_t *vol_obj,
-                 const H5VL_loc_params_t *loc_params, hid_t lcpl_id, hid_t lapl_id, hid_t dxpl_id, void **req,
-                 ...)
+H5VL_link_create(H5VL_link_create_type_t create_type, const H5VL_link_create_args_t *create_args,
+                 const H5VL_object_t *vol_obj, const H5VL_loc_params_t *loc_params, hid_t lcpl_id,
+                 hid_t lapl_id, hid_t dxpl_id, void **req)
 {
     H5VL_object_t tmp_vol_obj;               /* Temporary object */
-    va_list       arguments;                 /* Argument list passed from the API call */
-    hbool_t       arg_started     = FALSE;   /* Whether the va_list has been started */
     hbool_t       vol_wrapper_set = FALSE;   /* Whether the VOL object wrapping context was set up */
     herr_t        ret_value       = SUCCEED; /* Return value */
 
     FUNC_ENTER_NOAPI(FAIL)
 
-    /* Start the varargs, so they can be copied */
-    HDva_start(arguments, req);
-    arg_started = TRUE;
-
     /* Special case for hard links */
     if (H5VL_LINK_CREATE_HARD == create_type && NULL == vol_obj->data) {
-        va_list tmp_arguments; /* Copy of argument list passed in */
-
         /* Get the VOL data pointer from the varargs */
-        HDva_copy(tmp_arguments, arguments);
```

Comment:
An example of a case where avoiding var args can make things easier to unpack, as we don't need to worry about situations like va_copy to avoid interfering with the va_start above.